### PR TITLE
Issues/114 soap client

### PR DESCRIPTION
**Pull request by kvakos**
jaxws client

### CODE REVIEW

File: powerauth-java-client-jaxws/pom.xml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>powerauth-java-client-jaxws</artifactId>
+    <version>0.19.0</version>
+    <name>powerauth-java-client-jaxws</name>
+    <description>PowerAuth Server SOAP Service JaxWS Client</description>
```
Comment (romanstrobl) [inline powerauth-java-client-jaxws/pom.xml]: Let's use the official project name: `JAX-WS`, so the description would be `PowerAuth Server SOAP Service JAX-WS Client`.

File: powerauth-java-client-jaxws/src/main/java/io/getlime/powerauth/soap/adapter/DateTimeToLocalDateTime.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
+package io.getlime.powerauth.soap.adapter;
```
Comment (romanstrobl) [inline powerauth-java-client-jaxws/src/main/java/io/getlime/powerauth/soap/adapter/DateTimeToLocalDateTime.java]: Please add copyright.

File: powerauth-java-client-jaxws/src/main/resources/binding.xjb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<jaxb:bindings xmlns:xs="http://www.w3.org/2001/XMLSchema"
+               xmlns:jaxb="http://java.sun.com/xml/ns/jaxb" xmlns:xjc="http://java.sun.com/xml/ns/jaxb/xjc"
+               jaxb:extensionBindingPrefixes="xjc" version="2.1">
+    <jaxb:globalBindings typesafeEnumMemberName="generateName" generateIsSetMethod="false"
+                         generateElementProperty="false">
+        <xjc:simple/>
+        <xjc:javaType adapter="io.getlime.powerauth.soap.adapter.DateTimeToLocalDateTime"
+                      name="java.time.LocalDateTime" xmlType="xs:dateTime"/>
+        <!--<xjc:javaType adapter="io.getlime.powerauth.soap.adapter.DateToLocalDate"
```
Comment (romanstrobl) [inline powerauth-java-client-jaxws/src/main/resources/binding.xjb]: I suggest to remove the commented out adapters, if there is no plan to use them.

File: powerauth-java-client-jaxws/src/test/java/io/getlime/powerauth/soap/adapter/DateTimeToLocalDateTimeBerlinTest.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package io.getlime.powerauth.soap.adapter;
```
Comment (romanstrobl) [inline powerauth-java-client-jaxws/src/test/java/io/getlime/powerauth/soap/adapter/DateTimeToLocalDateTimeBerlinTest.java]: Please add missing copyright.

File: powerauth-java-client-jaxws/src/test/java/io/getlime/powerauth/soap/adapter/DateTimeToLocalDateTimeUtcTest.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
+package io.getlime.powerauth.soap.adapter;
```
Comment (romanstrobl) [inline powerauth-java-client-jaxws/src/test/java/io/getlime/powerauth/soap/adapter/DateTimeToLocalDateTimeUtcTest.java]: Please add missing copyright.

File: powerauth-java-client-jaxws/src/test/java/io/getlime/powerauth/soap/adapter/ZoneRule.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package io.getlime.powerauth.soap.adapter;
```
Comment (romanstrobl) [inline powerauth-java-client-jaxws/src/test/java/io/getlime/powerauth/soap/adapter/ZoneRule.java]: Please add missing copyright.

**Comment by romanstrobl** [comment]
The following error occurs when compiling the JAX-WS client on Java 11:
```
[WARNING] The POM for com.sun.xml.ws:jaxws-tools:jar:2.2.10 is invalid, transitive dependencies (if any) will not be available, enable debug logging for more details
[INFO] Processing: file:/.../powerauth-java-client-jaxws/src/main/resources/soap/wsdl/service.wsdl
[INFO] jaxws:wsimport args: [-keep, -s, '/.../powerauth-java-client-jaxws/target/generated-sources/wsimport', -d, '/.../powerauth-java-client-jaxws/target/classes', -encoding, UTF-8, -Xnocompile, -p, io.getlime.powerauth.soap, -b, 'file:/.../powerauth-java-client-jaxws/src/main/resources/binding.xjb', "file:/.../powerauth-java-client-jaxws/src/main/resources/soap/wsdl/service.wsdl"]
Error: Could not create the Java Virtual Machine.
-Xbootclasspath/p is no longer a supported option.
Error: A fatal exception has occurred. Program will exit.
```

**Comment by romanstrobl** [comment]
I will try to look for a solution to compile the project on Java 11.

**Comment by romanstrobl** [comment]
Based on this comment https://github.com/mojohaus/jaxws-maven-plugin/pull/71#issuecomment-440277459 the PR https://github.com/mojohaus/jaxws-maven-plugin/pull/71 works. Hopefully it will be merged soon.

**Comment by petrdvorak** [comment]
Let's migrate to REST API instead of introducing a new technology for Web Services.

**Pull request closed without merging**